evm-contracts/src/v0.6/dev/AggregatorFacade.sol

```diff
@@ -0,0 +1,174 @@
+   * @dev Note that for in-progress rounds (i.e. rounds that haven't yet received
+   * maxSubmissions) answer and updatedAt may change between queries.
+   */
+  function latestRoundData()
+    external
+    view
+    virtual
+    override
+    returns (
+      uint256 roundId,
+      int256 answer,
+      uint256 startedAt,
+      uint256 updatedAt,
+      uint256 answeredInRound
+    )
+  {
+    return _getRoundData(aggregator.latestRound());
+  }
+
+  /**
+   * @notice get past rounds answers
+   * @param _roundId the answer number to retrieve the answer for
+   */
+  function getAnswer(uint256 _roundId)
+    external
+    view
+    virtual
+    override
+    returns (int256)
+  {
+    return aggregator.getAnswer(_roundId);
+  }
+
+  /**
+   * @notice get block timestamp when an answer was last updated
+   * @param _roundId the answer number to retrieve the updated timestamp for
+   */
+  function getTimestamp(uint256 _roundId)
+    external
+    view
+    virtual
+    override
+    returns (uint256)
+  {
+    return aggregator.getTimestamp(_roundId);
+  }
+
+  /**
+   * @notice get all details about a round. Consumers are encouraged to check
+   * that they're receiving fresh data by inspecting the updatedAt and
+   * answeredInRound return values.
+   * @param _roundId the round ID to retrieve the details for. If _roundId
+   * has the special value UINT256_MAX (2**256-1), the contract will retrieve
+   * the latest round's details.
+   * @return roundId is the round ID for which details were retrieved
+   * @return answer is the answer for the given round
+   * @return startedAt is the timestamp when the round was started. This is 0
+   * if the round hasn't been started yet.
+   * @return updatedAt is the timestamp when the round last was updated (i.e.
+   * answer was last computed)
+   * @return answeredInRound is the round ID of the round in which the answer
+   * was computed. answeredInRound may be smaller than roundId when the round
+   * timed out. answerInRound is equal to roundId when the round didn't time out
+   * and was completed regularly.
+   * @dev Note that for in-progress rounds (i.e. rounds that haven't yet received
+   * maxSubmissions) answer and updatedAt may change between queries.
+   */
+  function getRoundData(uint256 _roundId)
+    external
+    view
+    virtual
+    override
+    returns (
+      uint256 roundId,
+      int256 answer,
+      uint256 startedAt,
+      uint256 updatedAt,
+      uint256 answeredInRound
+    )
+  {
+    return _getRoundData(_roundId);
+  }
+
+  /*
+   * Internal
+   */
+
+  function _getRoundData(uint256 _roundId)
+    internal
+    view
+    returns (
+      uint256 roundId,
+      int256 answer,
+      uint256 startedAt,
+      uint256 updatedAt,
+      uint256 answeredInRound
+    )
+  {
+    answer = aggregator.getAnswer(_roundId);
+    updatedAt = uint64(aggregator.getTimestamp(_roundId));
+    return (_roundId, answer, 0, updatedAt, _roundId);
```

Comment:
In the FluxAggregator, `startedAt == 0` is used to signal that the round has not started. I'd suggest setting it to updatedAt or 1 instead.

Reply:
Done